src/test/javascript/testBookScript.js

```diff
@@ -81,3 +81,13 @@ describe('Testing common.js - parseQueryString', () => {
     expect(result).toEqual({});
   });
 });
+
+describe('Testing common.js - getBookJSON', () => {
+  it('getting the correct info about a book', async () => {
+    const result = await getBookJSON('OBM3AAAAIAAJ');
```

Comment:
In general, unit tests should be hermetic, that means they don't require external dependencies. This unit test is actually calling into the real Google Books API, so any issue with the API availability will result in your tests flaking.

What you would generally do here is mock the Google Books API. That's like saying, I don't care what the real Google Books API actually returns, but assuming it returns this, getBookJSON will behave in this way. Not sure what's the best way to do this, but  http://www.wheresrhys.co.uk/fetch-mock/ seems promising.

That said, this function doesn't really do anything other than a fetch request, so it wouldn't actually be testing anything worth testing. This trick could be useful though if you want to be able to test any sort of processing of the google books api response.

Reply:
I see, I will just give up on it then.